File: src/main/java/com/youngjun/auth/core/domain/token/TokenWriter.java
package com.youngjun.auth.core.domain.token;

import jakarta.transaction.Transactional;
import org.springframework.stereotype.Component;

@Component
public class TokenWriter {

    private final TokenRepository tokenRepository;

    public TokenWriter(TokenRepository tokenRepository) {
        this.tokenRepository = tokenRepository;
    }

    @Transactional
    public void replaceTo(TokenPair tokenPair) {
        tokenRepository.delete(tokenPair.getAuthId());
        tokenRepository.write(tokenPair);
    }

}

File: src/main/kotlin/com/youngjun/auth/core/domain/token/TokenWriter.kt
package com.youngjun.auth.core.domain.token

import com.youngjun.auth.storage.db.core.token.TokenRepository
import jakarta.transaction.Transactional
import org.springframework.stereotype.Component

@Component
class TokenWriter(
    private val tokenRepository: TokenRepository,
) {
    @Transactional
    fun replaceTo(tokenPair: TokenPair) {
        tokenRepository.delete(tokenPair.userId)
        tokenRepository.write(tokenPair)
    }
}
